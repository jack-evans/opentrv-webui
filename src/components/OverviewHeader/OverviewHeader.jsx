import React from 'react'
import { Tabs, Tab } from 'carbon-components-react'

function OverviewHeader ({ tabSelected, handleTabChange }) {
  return (
    <div className='OverviewHeader'>
      <div className='OverviewHeader__title'>
        <h1>Overview</h1>
      </div>
      <div className='OverviewHeader__tabs-container'>
        <div className='OverviewHeader__tabs'>
          <Tabs selected={tabSelected} onSelectionChange={(event) => handleTabChange(event)}>
            <Tab className='OverviewHeader__tab' label='Devices' />
            <Tab className='OverviewHeader__tab' label='Policies' />
          </Tabs>
        </div>
      </div>
      <div className='OverviewHeader__title' />
    </div>
  )
}

export default OverviewHeader
