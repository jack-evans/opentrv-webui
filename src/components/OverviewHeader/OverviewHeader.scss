
.OverviewHeader {
  background-color: $ui-04;
  display: flex;
  height: 70px;

  .OverviewHeader__title {
    align-items: center;
    display: flex;
    max-width: 200px;
    padding: 20px 0 20px 40px;
    width: 100%;
  }

  .OverviewHeader__tabs-container {
    display: flex;
    justify-content: center;
    width: 100%;

    .OverviewHeader__tabs {
      display: flex;
      justify-content: flex-start;
      margin-top: 20px;
      width: 250px;

      .OverviewHeader__tab {
        font-size: 18pt;
      }
    }
  }
}