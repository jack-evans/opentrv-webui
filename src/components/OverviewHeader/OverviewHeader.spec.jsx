import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import OverviewHeader from './OverviewHeader'

Enzyme.configure({ adapter: new Adapter() })

describe('Overview.jsx', () => {
  it('renders without crashing', () => {
    expect(shallow(<OverviewHeader />)).toMatchSnapshot()
  })
})
