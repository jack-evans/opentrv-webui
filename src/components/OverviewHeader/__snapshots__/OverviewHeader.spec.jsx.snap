// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview.jsx renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OverviewHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="OverviewHeader__title"
        >
          <h1>
            Overview
          </h1>
        </div>,
        <div
          className="OverviewHeader__tabs-container"
        >
          <div
            className="OverviewHeader__tabs"
          >
            <Tabs
              ariaLabel="listbox"
              href="#"
              iconDescription="show menu options"
              onSelectionChange={[Function]}
              role="navigation"
              selected={0}
              triggerHref="#"
            >
              <Tab
                className="OverviewHeader__tab"
                href="#"
                label="Devices"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="presentation"
                selected={false}
                tabIndex={0}
              />
              <Tab
                className="OverviewHeader__tab"
                href="#"
                label="Policies"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="presentation"
                selected={false}
                tabIndex={0}
              />
            </Tabs>
          </div>
        </div>,
        <div
          className="OverviewHeader__title"
        />,
      ],
      "className": "OverviewHeader",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Overview
          </h1>,
          "className": "OverviewHeader__title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Overview",
          },
          "ref": null,
          "rendered": "Overview",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="OverviewHeader__tabs"
          >
            <Tabs
              ariaLabel="listbox"
              href="#"
              iconDescription="show menu options"
              onSelectionChange={[Function]}
              role="navigation"
              selected={0}
              triggerHref="#"
            >
              <Tab
                className="OverviewHeader__tab"
                href="#"
                label="Devices"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="presentation"
                selected={false}
                tabIndex={0}
              />
              <Tab
                className="OverviewHeader__tab"
                href="#"
                label="Policies"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="presentation"
                selected={false}
                tabIndex={0}
              />
            </Tabs>
          </div>,
          "className": "OverviewHeader__tabs-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Tabs
              ariaLabel="listbox"
              href="#"
              iconDescription="show menu options"
              onSelectionChange={[Function]}
              role="navigation"
              selected={0}
              triggerHref="#"
            >
              <Tab
                className="OverviewHeader__tab"
                href="#"
                label="Devices"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="presentation"
                selected={false}
                tabIndex={0}
              />
              <Tab
                className="OverviewHeader__tab"
                href="#"
                label="Policies"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="presentation"
                selected={false}
                tabIndex={0}
              />
            </Tabs>,
            "className": "OverviewHeader__tabs",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "ariaLabel": "listbox",
              "children": Array [
                <Tab
                  className="OverviewHeader__tab"
                  href="#"
                  label="Devices"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="presentation"
                  selected={false}
                  tabIndex={0}
                />,
                <Tab
                  className="OverviewHeader__tab"
                  href="#"
                  label="Policies"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="presentation"
                  selected={false}
                  tabIndex={0}
                />,
              ],
              "href": "#",
              "iconDescription": "show menu options",
              "onSelectionChange": [Function],
              "role": "navigation",
              "selected": 0,
              "triggerHref": "#",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "OverviewHeader__tab",
                  "href": "#",
                  "label": "Devices",
                  "onClick": [Function],
                  "onKeyDown": [Function],
                  "role": "presentation",
                  "selected": false,
                  "tabIndex": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "OverviewHeader__tab",
                  "href": "#",
                  "label": "Policies",
                  "onClick": [Function],
                  "onKeyDown": [Function],
                  "role": "presentation",
                  "selected": false,
                  "tabIndex": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "OverviewHeader__title",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="OverviewHeader__title"
          >
            <h1>
              Overview
            </h1>
          </div>,
          <div
            className="OverviewHeader__tabs-container"
          >
            <div
              className="OverviewHeader__tabs"
            >
              <Tabs
                ariaLabel="listbox"
                href="#"
                iconDescription="show menu options"
                onSelectionChange={[Function]}
                role="navigation"
                selected={0}
                triggerHref="#"
              >
                <Tab
                  className="OverviewHeader__tab"
                  href="#"
                  label="Devices"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="presentation"
                  selected={false}
                  tabIndex={0}
                />
                <Tab
                  className="OverviewHeader__tab"
                  href="#"
                  label="Policies"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="presentation"
                  selected={false}
                  tabIndex={0}
                />
              </Tabs>
            </div>
          </div>,
          <div
            className="OverviewHeader__title"
          />,
        ],
        "className": "OverviewHeader",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Overview
            </h1>,
            "className": "OverviewHeader__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Overview",
            },
            "ref": null,
            "rendered": "Overview",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="OverviewHeader__tabs"
            >
              <Tabs
                ariaLabel="listbox"
                href="#"
                iconDescription="show menu options"
                onSelectionChange={[Function]}
                role="navigation"
                selected={0}
                triggerHref="#"
              >
                <Tab
                  className="OverviewHeader__tab"
                  href="#"
                  label="Devices"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="presentation"
                  selected={false}
                  tabIndex={0}
                />
                <Tab
                  className="OverviewHeader__tab"
                  href="#"
                  label="Policies"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="presentation"
                  selected={false}
                  tabIndex={0}
                />
              </Tabs>
            </div>,
            "className": "OverviewHeader__tabs-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Tabs
                ariaLabel="listbox"
                href="#"
                iconDescription="show menu options"
                onSelectionChange={[Function]}
                role="navigation"
                selected={0}
                triggerHref="#"
              >
                <Tab
                  className="OverviewHeader__tab"
                  href="#"
                  label="Devices"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="presentation"
                  selected={false}
                  tabIndex={0}
                />
                <Tab
                  className="OverviewHeader__tab"
                  href="#"
                  label="Policies"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="presentation"
                  selected={false}
                  tabIndex={0}
                />
              </Tabs>,
              "className": "OverviewHeader__tabs",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "ariaLabel": "listbox",
                "children": Array [
                  <Tab
                    className="OverviewHeader__tab"
                    href="#"
                    label="Devices"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="presentation"
                    selected={false}
                    tabIndex={0}
                  />,
                  <Tab
                    className="OverviewHeader__tab"
                    href="#"
                    label="Policies"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="presentation"
                    selected={false}
                    tabIndex={0}
                  />,
                ],
                "href": "#",
                "iconDescription": "show menu options",
                "onSelectionChange": [Function],
                "role": "navigation",
                "selected": 0,
                "triggerHref": "#",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "OverviewHeader__tab",
                    "href": "#",
                    "label": "Devices",
                    "onClick": [Function],
                    "onKeyDown": [Function],
                    "role": "presentation",
                    "selected": false,
                    "tabIndex": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "OverviewHeader__tab",
                    "href": "#",
                    "label": "Policies",
                    "onClick": [Function],
                    "onKeyDown": [Function],
                    "role": "presentation",
                    "selected": false,
                    "tabIndex": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "OverviewHeader__title",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
